src/opm/io/eclipse/EclOutput.cpp

```diff
@@ -91,12 +91,53 @@ void EclOutput::flushStream()
     this->ofileH.flush();
 }
 
-void EclOutput::writeBinaryHeader(const std::string&arrName, int size, eclArrType arrType)
+void EclOutput::writeArrayType(const eclArrType arrType)
 {
+    switch(arrType) {
+    case INTE:
+        ofileH.write("INTE", 4);
+        break;
+    case REAL:
+        ofileH.write("REAL", 4);
+        break;
+    case DOUB:
+        ofileH.write("DOUB", 4);
+        break;
+    case LOGI:
+        ofileH.write("LOGI", 4);
+        break;
+    case CHAR:
+        ofileH.write("CHAR", 4);
+        break;
+    case MESS:
+        ofileH.write("MESS", 4);
+        break;
+    }
+}
+
+void EclOutput::writeBinaryHeader(const std::string&arrName, long int size, eclArrType arrType)
+{
+    int bhead = flipEndianInt(16);
     std::string name = arrName + std::string(8 - arrName.size(),' ');
 
+    // write X231 header if size larger that limits for 4 byte integers
+
+    if (size > std::numeric_limits<int>::max()) {
+        long int val231 = pow(2,31);
```

Comment:
I guess this could be written `val231 = 2 << 31`; if not I think you should prefix the `pow` with `std::`